jbmc/src/java_bytecode/java_entry_point.cpp

```diff
@@ -88,7 +127,53 @@ static void java_static_lifetime_init(
     const symbolt &sym=*symbol_table.lookup(symname);
     if(should_init_symbol(sym))
     {
-      if(sym.value.is_nil() && sym.type!=empty_typet())
+      if(const symbolt *class_literal_init_method =
+         get_class_literal_initializer(sym, symbol_table))
+      {
+        const std::string &name_str = id2string(sym.name);
+        irep_idt class_name =
+          name_str.substr(0, name_str.size() - strlen("@class_model"));
+        const symbolt &class_symbol = symbol_table.lookup_ref(class_name);
+
+        bool class_is_array = is_java_array_tag(sym.name);
+
+        exprt name_literal(ID_java_string_literal);
+        name_literal.set(ID_value, to_class_type(class_symbol.type).get_tag());
+
+        symbol_exprt class_name_literal =
+          get_or_create_string_literal_symbol(
+            name_literal, symbol_table, string_refinement_enabled);
+
+        // Call the literal initializer method instead of a nondet initializer:
+
+        // For arguments we can't parse yet:
+        side_effect_expr_nondett nondet_bool(java_boolean_type());
+
+        // Argument order is: name, isAnnotation, isArray, isInterface,
+        // isSynthetic, isLocalClass, isMemberClass, isEnum
+
+        code_function_callt initializer_call;
+        initializer_call.function() = class_literal_init_method->symbol_expr();
+
+        code_function_callt::argumentst &args = initializer_call.arguments();
+
+        args.push_back(address_of_exprt(sym.symbol_expr())); /* this */
+        args.push_back(address_of_exprt(class_name_literal));
```

Comment:
Comment on the end for each one would be great (rather than at the top) so don't have to cross reference